src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -396,6 +400,9 @@ public function formAction($id, Request $request)
 
         $productAdapter = $this->get('prestashop.adapter.data_provider.product');
         $product = $productAdapter->getProduct($id);
+        if (Context::getContext()->shop->getContext() == Shop::CONTEXT_SHOP && !$product->isAssociatedToShop()) {
```

Comment:
imo this is the responsibility of productAdapter to return the "right" new product.

As a proof, we can see a code smell: we affect $product variable twice.

Reply:
That was also my first thought, but the ProductAdapter is called from several places, and I'm not sure we want this logic everywhere: it should applies only in the ProductController